parser/src/string.rs

```diff
@@ -322,25 +315,27 @@ impl<'a> StringParser<'a> {
                                 ),
                                 conversion,
                                 format_spec: spec,
-                                range: self.range(),
+                                range: self.current_range(location),
                             }
                             .into(),
                         )]
                     } else {
+                        // TODO: range is wrong but `self_documenting` needs revisiting beyond
+                        // ranges
```

Comment:
Can you add a link to the issue. I think it would be good to fix the ranges as part of this PR, and potentially extend it in another PR

Reply:
not linked in the code (will add) but linked from a comment here. i believe all the oddness in the snapshots you see are the same issue, which i raised in https://github.com/astral-sh/ruff/issues/5970

maybe i don't explain it well enough but the same source `{foo=}` is generating _multiple nodes (including the possibly empty one) 

fixing it requires a design decision (or just help) from that other issue

Reply:
Okay. Let me take a look at the grammar to fully understand the issue and what needs to be parsed here.

Reply:
I replied on the linked issue: https://github.com/astral-sh/ruff/issues/5970